ably/realtime_channel.go

```diff
@@ -167,9 +167,37 @@ func (c *RealtimeChannel) mayAttach(result, checkActive bool) (Result, error) {
 			return nopResult, nil
 		}
 	}
-	if !c.client.Connection.lockIsActive() {
-		return nil, c.state.set(StateChanFailed, errAttach)
+
+	switch c.client.Connection.State() {
+	// RTL4b
+	case StateConnClosed,
+		StateConnClosing,
+		StateConnFailed:
+		return nil, newError(80000, errAttach)
+
+	// RTL4i
```

Comment:
can't find this on the spec.

Reply:
[RTL4i](https://docs.ably.io/client-lib-development-guide/features/#RTL4i)